azure-kusto-ingest/tests/test_kusto_ingest_client.py

```diff
@@ -0,0 +1,123 @@
+"""Test for KustoIngestClient."""
+
+import os
+import json
+import unittest
+from mock import patch
+import base64
+import re
+
+from azure.kusto.data import KustoClient, KustoServiceError
+from azure.kusto.ingest import KustoIngestClient, IngestionProperties, DataFormat
+
+
+UUID_REGEX = "[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"
+BLOB_NAME_REGEX = "database__table__" + UUID_REGEX + "__dataset.csv.gz"
+BLOB_URL_REGEX = (
+    "https://storageaccount.blob.core.windows.net/tempstorage/database__table__"
+    + UUID_REGEX
+    + "__dataset.csv.gz[?]sas"
+)
+
+
+def mocked_aad_helper(*args, **kwargs):
+    """Mock to replace _AadHelper._acquire_token"""
+    return None
+
+
+def mocked_requests_post(*args, **kwargs):
+    """Mock to replace requests.post"""
+
+    class MockResponse:
+        """Mock class for KustoResponse."""
+
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.text = str(json_data)
```

Comment:
https://stackoverflow.com/questions/18034272/python-str-vs-unicode-types

This is why we use six.
You can look at the KustoClient tests as an example

Reply:
Changing  here and also in other places that 'str' appears in our code.